src/job.rs

```diff
@@ -690,7 +690,7 @@ impl Job {
                     }
                     Ok(c) => c,
                 };
-                if chat.typ == Chattype::Group {
+                if chat.typ == Chattype::Group || chat.typ == Chattype::Mailinglist {
```

Comment:
I would use `match` here with an explicit no-op branch for all other variants.

Also `Message::load_from_db` can be postponed until we know it's a group or mailinglist, so we don't load the messages we don't need from the database.

Reply:
we're using a `match` now.

for postponing ` Message::load_from_db`: it seems we need the message-object to get the chat-object and finally the chattype. so, i concentaed on the `match` only and avoided larger refactoring.